core/kazoo_auth/src/kz_auth_client.erl

```diff
@@ -304,3 +315,58 @@ add_claims(Claims) ->
 
 add_claim(<<"iat">>) -> {iat, kz_time:current_unix_tstamp()-500};
 add_claim(<<"exp">>) -> {exp, kz_time:current_unix_tstamp()+(2 * ?MILLISECONDS_IN_SECOND)}.
+
+-spec request(kz_term:ne_binary(), map()) -> kz_http:ret().
+request(URL, Token) ->
+    request('get', URL, <<>>, Token).
+
+-spec request(atom(), kz_term:ne_binary(), binary(), map()) -> {ok|error, any(), kz_term:proplist()}.
+request(Verb, URL, Body, #{token := #{authorization := Authorization}}) ->
+    Options = [{headers_as_is, true}, {ssl, [{versions, ['tlsv1.2']}]}],
+    {ok,{_,_, Host, _, _, _}} = http_uri:parse(URL),
+    Headers = [{<<"host">>, Host}
+              ,{<<"Content-Type">>, <<"application/json">>}
+              ,{"Authorization", kz_term:to_list(Authorization)}
+              ],
+    case kz_http:req(Verb, URL, Headers, Body, Options) of
+        {'ok', 200, _RespHeaders, RespBody} -> {ok, RespBody};
+        {'ok', 401, _RespHeaders, _RespXML} -> {error, 'not_authorized'};
+        {'ok', 404, _RespHeaders, _RespXML} -> {error, 'not_found'};
+        {'ok', _Code, _RespHeaders, _RespXML} -> {error, 'other'};
+        {'error', _}=Error -> Error
+    end.
+
+-spec cached_token(map()) -> map().
+cached_token(#{subject := #{'_id' := Id}} = Map) ->
+    case kz_cache:fetch_local(?TOKENS_CACHE, {client, Id}) of
```

Comment:
I don't follow why the 'expires_absolute' flag is needed when `kz_cache:peek_local` takes care of not bumping the timestamp used for expiration. Can you not use `peek_local` here instead of making the change to kazoo_caches?